src/Xamarin.Android.Build.Tasks/Tasks/BuildArchive.cs

```diff
@@ -0,0 +1,253 @@
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Android.Build.Tasks;
+using Microsoft.Build.Framework;
+using Xamarin.Tools.Zip;
+
+namespace Xamarin.Android.Tasks;
+
+/// <summary>
+/// Takes a list of files and adds them to an APK archive. If the APK archive already
+/// exists, files are only added if they were changed.
+/// </summary>
+public class BuildArchive : AndroidTask
+{
+	public override string TaskPrefix => "BAA";
+
+	public string? AndroidPackageFormat { get; set; }
+
+	public string? ApkInputPath { get; set; }
+
+	[Required]
+	public string ApkOutputPath { get; set; } = null!; // NRT enforced by [Required]
```

Comment:
Sometimes for these I just put:
```suggestion
	public string ApkOutputPath { get; set; } = "";
```